### PR TITLE
feat: wire extended-hours counter-trading (hedge, position-check, conductor)

### PR DESCRIPTION
**Pull request by JuaniRios**
## What

- Wire the extended-hours policy on top of the aggregate: session-aware hedge placement (`apply_slippage` limit pricing with Alpaca penny/sub-penny rounding, Market-vs-`ExtendedHoursLimit` selection by `MarketSession` re-checked at perform time), the level-triggered cancel-and-replace sweep that converges still-live extended-hours limit orders to market orders at the regular open, and the inline extended-hours enqueue path.
- Replace the conservative placeholders from #916/#931 with the real logic: `hedge` (in `PlaceHedge::perform`) derives `CounterTradeOrderKind` from the session re-checked at perform time, while `conductor` keeps inline regular-hours placement as `Market` and, for an extended-hours fill, enqueues an immediate `PlaceHedge` job instead of placing inline. Cancellations route through the shared finalization helpers.

## Why

- Final implementation PR of the stack (#915 -> #916 -> #931 -> **#932**). Turns the state machine + execution primitives into the actual extended-hours behaviour. Ships inert (config `"disabled"` everywhere) until enabled per asset in a later config flip.

## How

- `src/trading/offchain/hedge.rs`: session recheck at perform time (run unconditionally — see the post-review fix below), `apply_slippage`, order-kind selection with a per-symbol extended-hours gate, submission lock.
- `src/position_check.rs`: `request_extended_hours_cancellations` sweep gated by `is_extended_hours_enabled`, plus `finalize_terminal_pending_positions`.
- `src/conductor.rs` + `src/conductor/builder.rs`: inline extended-hours enqueue, the hedge `OrderPlacer` wired as a session source, `hedge_queue` wiring, and orphan recovery routed through the shared `position_command_for_finalization` helper.
- `src/onchain/accumulator.rs`, `src/trading/onchain/trade_accountant.rs`, `src/conductor/monitor/executor_maintenance.rs`, `src/integration_tests/arbitrage.rs`.

## Testing

- Unit tests for session-aware order-kind selection / `apply_slippage`, the cancel-and-replace sweep, and the conductor extended-hours enqueue path (`extended_hours_trade_enqueues_immediate_hedge_job`); hedge tests cover both decoupling failure modes (a stale Regular job skips when the live venue is closed; a disabled-symbol job skips during an Extended session) and the terminal-pending recovery no-re-poll path; `integration_tests/arbitrage.rs` updated. Verified locally on the stack tip: `cargo check --all-targets`, `cargo clippy --all-features`, `cargo fmt --check` clean. CI runs the full suite per PR.

## Anything else

- Post-review blocker fix ([RAI-1173](https://linear.app/makeitrain/issue/RAI-1173)): the perform-time market-session recheck was gated by the global `order_placer` flag, which was `Some` only when some asset enabled extended hours. That let a stale Regular job submit a market order after the venue closed (when all assets were disabled) or place an orphan-able extended-hours limit for a per-symbol-disabled symbol. Fix: the `OrderPlacer` is now **always** constructed (a pure session source, decoupled from the feature flag), so the recheck always runs; per-symbol eligibility is carried in `HedgeCtx.assets` and the `Extended` arm skips symbols that are not enabled. `recover_pending_poll_status` now warns instead of silently returning `Ok` for a terminal pending order, and the unreachable `OrderPlacerNotConfigured` error was removed.
- Inert until enabled per asset: with every asset `"disabled"`, the always-present `OrderPlacer` never reaches a limit-order placement because the per-symbol `is_extended_hours_enabled` gate skips every Extended placement, and the cancel-and-replace sweep (`request_extended_hours_cancellations`) skips every symbol — so behaviour is unchanged (market orders during regular hours only).
- One unconditional behaviour change: the `CheckPositions` tick now always runs `finalize_terminal_pending_positions` (finalizes terminal pending offchain orders into position commands, with `Cancelled` orders finalized via cancel/complete rather than failure semantics) regardless of the flag — active even with the feature disabled.
- This stack applies the extended-hours feature on top of current master and deliberately preserves master's operator-recovery / equity-redemption (schema 4) work -- the original single-PR branch predated those and this stack does not revert them.
- Stack: #915 -> #916 -> #931 -> **#932**.

## Summary by CodeRabbit

* **New Features**
  * Added extended-hours session-aware trade handling, including slippage-aware counter-trading with limit orders and automatic hedge job routing.
  * Enhanced periodic position monitoring to finalize terminal pending states and sweep/cancel still-live extended-hours orders when enabled.

* **Bug Fixes**
  * Improved startup/orphan order recovery to avoid silent undecided states and to ensure cancelled orders and unpriced terminal partial fills are handled consistently.
  * Adjusted cancelled-state dispatch so it no longer surfaces as a generic failure during post-placement processing.

* **Tests**
  * Updated and expanded coverage for extended-hours hedging, recovery, cancellation routing, and retry behavior.

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/932?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#932** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/932?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#917](https://github.com/ST0x-Technology/st0x.liquidity/pull/917) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/917?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#941](https://github.com/ST0x-Technology/st0x.liquidity/pull/941) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/941?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/932?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#931** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/931?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#916** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/916?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#915** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/915?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=932).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by JuaniRios** [comment]
@coderabbitai full review

File: src/conductor.rs [hunk 1]
```diff
@@ -2173,6 +2195,57 @@ where
...
+    // OrderPlacer for the limit-order price lookup). Instead of waiting for the
+    // next CheckPositions scan (~1 min), enqueue an immediate PlaceHedge job,
+    // which has the OrderPlacer and submits the limit order. CheckPositions
+    // remains the backstop. Preflight + clamp here, mirroring the scan path, so
+    // we never enqueue a hedge that would exceed buying power.
+    if execution.market_session == MarketSession::Extended {
+        let _counter_trade_submission_guard = cqrs.counter_trade_submission_lock.lock().await;
+
+        match preflight_counter_trade_submission(executor, &execution, None).await? {
+            CounterTradeSubmissionCheck::Skipped => return Ok(None),
```
Comment (0xgleb) [inline src/conductor.rs:2531]: should fix: the new extended-hours block in process_queued_trade has its own preflight `Skipped => return Ok(None)` (this line) and `clamp_shares_to_reservation(&mut execution, ...)` just below, and the clamped `execution.shares` is what gets written into the enqueued PlaceHedge payload. The only test that reaches this branch (extended_hours_trade_enqueues_immediate_hedge_job) provisions 10 AAPL shares + $10k buying power against a 5-share hedge, so the preflight returns Allowed and the clamp never fires; it asserts job.shares == 5.0 un-clamped. The two skip tests (trade_above_threshold_skips_counter_trade_without_offchain_inventory, ..._still_skips_with_zero_inventory) run with a bare MockExecutor (Regular session), so they only hit the regular-path Skipped branch below, never this one. This is the financial-safety gate for the extended path and it's a separate copy of the regular branch that can drift. Please add two extended-session tests: one where the preflight skips (assert Ok(None), zero PlaceHedge jobs enqueued, position left unclaimed) and one where the reservation is smaller than the requested shares (assert the enqueued PlaceHedge.shares equals the clamped quantity).

File: src/trading/offchain/hedge.rs [hunk 2]
```diff
@@ -47,6 +137,12 @@ pub(crate) struct PlaceHedge {
     pub(crate) executor: SupportedExecutor,
     pub(crate) threshold: ExecutionThreshold,
     pub(crate) offchain_order_id: OffchainOrderId,
+    #[serde(default = "default_market_session")]
```
Comment (0xgleb) [inline src/trading/offchain/hedge.rs:138]: should fix: PlaceHedge is a durable apalis job persisted as JSON, and `#[serde(default = "default_market_session")]` is the entire mechanism that keeps in-flight jobs enqueued by the pre-deploy binary (which lack this field) deserializable after rollout. Right now nothing tests that path: the hedge.rs tests all build PlaceHedge via struct literal with the field set, and the only from_slice in the suite (the conductor enqueue test) round-trips a freshly-serialized job that already has the field. The codebase already has this exact kind of legacy-payload test elsewhere (rebalancing/equity/job.rs, offchain/order.rs, usdc_rebalance.rs). Add a small unit test that serde_json::from_value/from_str on a PlaceHedge JSON object lacking market_session succeeds and yields MarketSession::Regular, so a dropped attribute or wrong default variant (both of which compile clean) gets caught.

File: src/onchain/accumulator.rs [hunk 3]
```diff
@@ -76,20 +82,37 @@ fn check_asset_enabled(asset_enabled: bool, symbol: &Symbol) -> bool {
...
+    match session {
+        MarketSession::Regular => Ok(Some(session)),
+
+        MarketSession::Extended if extended_hours_enabled => {
+            debug!(target: "hedge", %symbol, "Extended hours session, will use limit order");
+            Ok(Some(session))
+        }
 
-    Ok(is_open)
+        MarketSession::Extended | MarketSession::Closed => {
```
Comment (0xgleb) [inline src/onchain/accumulator.rs:105]: should fix: this defer arm now folds in the new Extended-with-symbol-disabled case, which is reachable in production (trading enabled, extended_hours_counter_trading disabled, broker in an Extended session lands here and returns None). Every accumulator test uses MockExecutor::new() (Regular) or .with_market_closed() (Closed); none drives an Extended session, so the Closed subcase is covered but the Extended-disabled subcase is exercised nowhere (the Extended-enabled->Some path is only covered indirectly via the conductor test). If this arm were ever miswired to return Some, a disabled symbol would get hedged with a market order during extended hours. Add a check_execution_readiness test with MockExecutor::new().with_market_session(MarketSession::Extended) and extended_hours_counter_trading Disabled asserting None, and optionally a companion asserting Some(Extended) when enabled.

File: src/trading/offchain/hedge.rs [hunk 4]
```diff
@@ -209,7 +447,32 @@ impl Job<HedgeCtx> for PlaceHedge {
...
 
-            Some(Filled { .. } | Pending { .. } | Cancelled { .. }) | None => {}
+            // Cancelled here is unexpected -- a freshly Placed order should
+            // not be Cancelled by the time we re-load it. It is still a
+            // broker-confirmed cancellation, so route it through the shared
+            // terminal finalization: recording it as a failure would anchor
+            // the next attempt's client_order_id to the cancelled key (the
+            // broker would dedupe against a dead order) and drop any partial
+            // fill the cancellation retained.
+            Some(cancelled @ Cancelled { .. }) => {
```
Comment (0xgleb) [inline src/trading/offchain/hedge.rs]: should fix: this new arm routes a freshly-placed-then-Cancelled order through terminal_position_finalization + position_command_for_finalization, and the Complete path debits a retained partial fill from the position. The conductor twin dispatch_post_place_state got both a no-fill and a partial-fill Cancelled test in this PR, but the parallel copy here in PlaceHedge has none: every placer in the hedge suite yields Submitted (or Failed for the rejecting placer) after Place, never Cancelled, so this arm never runs. Since it's a hand-mirrored copy of the conductor logic that touches money, the two can drift undetected. Either drive a placer/aggregate setup that yields Cancelled at reload (asserting CompleteOffChainOrder debits net for a priced partial fill and CancelOffChainOrder for no-fill, mirroring the conductor tests), or factor the shared send-or-log wrapper so a single tested helper backs both sites (see the duplication note on the conductor arm).

File: src/conductor.rs [hunk 5]
```diff
@@ -2499,7 +2572,31 @@ async fn dispatch_post_place_state(
             Ok(None)
         }
 
-        Some(Pending { .. } | Filled { .. } | Cancelling { .. } | Cancelled { .. }) => {
+        // Cancelled directly after Place is unexpected but is a genuine
+        // broker-confirmed cancellation: route through the shared terminal
+        // finalization instead of recording a failure, which would anchor the
+        // next attempt's client_order_id to the cancelled key and drop any
+        // retained fill. Mirrors the matching arm in `PlaceHedge::perform`.
+        Some(cancelled @ Cancelled { .. }) => {
```
Comment (0xgleb) [inline src/conductor.rs:3074]: nit: this arm and the matching one in PlaceHedge::perform are the same ~12-line wrapper: `terminal_position_finalization(&cancelled).and_then(|f| position_command_for_finalization(f, id))`, then `if let Some(command) { position.send(...).await? } else { error!("...operator intervention required") }`, with a byte-identical operator-facing error string. The PR even self-documents it with "Mirrors the matching arm in PlaceHedge::perform." The money-critical decision (which command to emit) already lives in the two shared helpers, so this is low-stakes, but a small shared async helper (e.g. finalize_or_log_unpriced(position, symbol, id, &cancelled)) next to those helpers would keep the send + unpriced-fill error message in one place and let a single test cover both call sites.

File: src/trading/offchain/hedge.rs [hunk 6]
```diff
@@ -47,6 +137,12 @@ pub(crate) struct PlaceHedge {
     pub(crate) executor: SupportedExecutor,
     pub(crate) threshold: ExecutionThreshold,
     pub(crate) offchain_order_id: OffchainOrderId,
+    #[serde(default = "default_market_session")]
+    pub(crate) market_session: st0x_execution::MarketSession,
```
Comment (0xgleb) [inline src/trading/offchain/hedge.rs]: nit, two small things on this field. (1) MarketSession is already imported unqualified at the top of the file and used unqualified in the production match arms, so `st0x_execution::MarketSession` here (and in default_market_session's return type/body) is redundant prefix noise; drop it to match the rest of the file. (2) self.market_session is read in perform only for the staleness log comparison (current_session vs self.market_session); the operative session is always re-fetched into current_session and that is what derives order_kind, so this serialized field can never affect a trade decision. A one-line doc comment noting it's diagnostic-only would stop a future reader from assuming it gates the session-dependent order_kind logic.

File: src/trading/onchain/trade_accountant.rs [hunk 7]
```diff
@@ -237,6 +237,25 @@ pub(crate) enum TradeAccountingError {
...
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[error("Failed to fetch latest trade price for {symbol}")]
+    LimitPriceFetch {
+        symbol: st0x_execution::Symbol,
+        #[source]
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[error(
+        "Executor does not support fetching trade prices for \
```
Comment (0xgleb) [inline src/trading/onchain/trade_accountant.rs]: nit: this message split across two lines with a backslash continuation collapses to about 76 chars (`#[error("Executor does not support fetching trade prices for {symbol}")]`), well under the 100-char limit, and the neighboring MarketSessionCheck/LimitPriceFetch variants are single-line (MarketSessionCheck's message is actually longer). cargo fmt won't reformat literals inside #[error], so collapse it to one line for grep-ability and consistency.

**Review by 0xgleb — CHANGES_REQUESTED** [review]

**Review by 0xgleb — APPROVED** [review]
